server/api/changeProfilePicture.ts

```diff
@@ -0,0 +1,39 @@
+import { Request, Response } from "express";
+import { db } from "../database";
+import { sessions } from "../types/Session";
+
+/**
+ * Changes a user's username, given a valid session token and valid new password.
+ * @param {Request} req - Requests a JSON object in the body containing 
+ *      "token" and "username" field.
+ * @param {Response} res - Responds with error text in the case of a user or internal 
+ *      error (beginning with "Error:"), or with the newly generated session token in the case of
+ *      a success.
+ */
+
+export function changeProfilePicture(req: Request, res: Response) {
+    if ("token" in req.body && "profile_picture" in req.body) {
+
+        if (!sessions.has(req.body.token)) {
+            // invalid user token
+            console.log("[API] changeUsername failed! (invalid token provided)");
+            res.status(401).send("Error: Invalid token provided.");
+            return;
+        }
+
+        let oldUsername = sessions.get(req.body.token).username;
```

Comment:
Can be `const`, a little superfluous; you could move it to the `db.run` statement